examples/direct_fidelity_estimation.py

```diff
@@ -29,15 +31,19 @@ def build_circuit():
 
 def compute_characteristic_function(circuit: cirq.Circuit,
                                     P_i: Tuple[cirq.Gate, ...],
-                                    qubits: List[cirq.Qid],
-                                    noise: Optional[cirq.NoiseModel]):
+                                    qubits: List[cirq.Qid], physical: bool,
+                                    simulator: cirq.DensityMatrixSimulator,
+                                    density_matrix: Optional[np.ndarray]):
     n_qubits = len(P_i)
     d = 2**n_qubits
 
-    simulator = cirq.DensityMatrixSimulator()
-    # rho or sigma in https://arxiv.org/pdf/1104.3835.pdf
-    density_matrix = cast(cirq.DensityMatrixTrialResult,
-                          simulator.simulate(circuit)).final_density_matrix
+    if physical:
+        assert density_matrix is None
+        density_matrix = np.zeros([d, d], dtype=np.complex64)
+        res = simulator.run(circuit, repetitions=1).measurements['y']
+        # Convert binary to index.
+        idx = sum([res[0][i] * 2**i for i in range(len(res))])
+        density_matrix[idx][idx] = 1
```

Comment:
this isn't quite right

One of the points of DFE in contrast to regular tomography is you don't have to ever figure out the full density matrix. Using the expressions presented in the papers, you only need to know quantities of the form Tr(P_i rho) which you can estimate based on sampling using a scheme like that found in `PauliSumCollector`. 

The only complication is figuring out the value of `prob` (aka Pr(i)). The papers suggest that if you constrain your circuit to be only clifford operations, you can generate all (input) pauli strings and efficiently simulate how the circuit will transform them. Then you have a collection of output pauli strings uniformly distributed (i.e. Pr(i) = constant). For this PR, maybe it will be easier to still calculate `prob` via simulation; but the trace can be found by estimating operator expectation values

Reply:
Thanks. Doing Pr(i) estimation in simulation sounds good for now.

Added the call to PauliSumCollector. I wasn't sure whether I should use energy of its square root. I think it's without the square root, but not sure.

I also looked at the code just to understand what's happening, and it seems that another collector could be an actual, physical measurement, right?

Reply:
I don't see any changes (i.e. using pauli sum collector)

Reply:
Sorry about this. Let me try to upload again. I had added the Pauli sum collector.